src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -86,21 +86,24 @@ public void discoverClusterSlots(Jedis jedis) {
       rediscovering = true;
 
       try {
-        this.slots.clear();
-
         List<Object> slots = jedis.clusterSlots();
 
+        // We should clear slots after getting result about cluster slots, because if we got exception or timeout in
```

Comment:
It should be more stable behavior and makes sense. I will take a look how to implement it.